core/src/main/java/hudson/model/ListView.java

```diff
@@ -145,6 +147,9 @@ private Object readResolve() {
         }
         initColumns();
         initJobFilters();
+        if (statusFilter != null) {
+            jobFilters.add(new StatusFilter(statusFilter));
```

Comment:
This should add the `StatusFilter` first into the list to minimize behavior changes through interactions between filters.

Reply:
Rewritten to be added to the first index